File: api/Entities/HR/Candidate.cs
using System.ComponentModel.DataAnnotations;

namespace api.Entities.HR
{
    public class Candidate: BaseEntity
    {
        public int ApplicationNo { get; set; }
//names                
        [Required, MaxLength(1)]
        public string Gender {get; set;}="M";
        public string FirstName { get; set; }
        public string SecondName { get; set; }
        public string FamilyName { get; set; }
        //[Required]         
        public string KnownAs { get; set; }
        public int? CustomerId { get; set; }
        public string Source {get; set;}
        //[Required]
        public DateTime? DOB { get; set; }
        public string PpNo { get; set; } 
        public string AadharNo { get; set; }
        public string Ecnr { get; set; }
        public string Address {get; set;}
        public string City {get; set;}
        public string Pin {get; set;}
        public string Country { get; set; } = "India";
        public string Nationality {get; set;}="Indian";
        [EmailAddress]
        public string Email { get; set; }
    //general
        public DateTime Created { get; set; }
        public DateTime LastActive { get; set; }
        public int AppUserId {get; set;}
        public string Status {get; set;} = "NotReferred";
        public string Qualifications {get; set;}
        public ICollection<UserPhone> UserPhones {get; set;}
        public ICollection<UserProfession> UserProfessions {get; set;}
        public ICollection<UserExp> UserExperiences {get; set;}
        public ICollection<UserAttachment> UserAttachments {get; set;}
        public ICollection<UserQualification> UserQualifications {get; set;}
        public bool NotificationDesired {get; set;}
        public string Username { get; set; }
        public string FullName {get => FirstName + " " + SecondName + " " + FamilyName;}
    }
}

File: client/src/app/_models/hr/candidate.ts
import { IPhoto } from "../admin/photo";
import { IUserPhone } from "../params/Admin/userPhone";
import { IUserProfession } from "../params/Admin/userProfession";
import { IUserQualification } from "../params/Admin/userQualification";
import { IUserAttachment } from "./userAttachment";
import { IUserExp } from "./userExp";


export interface ICandidate {
     //checked: boolean;
     id: number;
     applicationNo: number;
     gender: string;
     firstName: string;
     secondName: string;
     familyName: string;
     knownAs: string;
     customerId?: number;
     referredByName: string;
     dOB?: Date;
     placeOfBirth: string;
     aadharNo: string;
     ppNo: string;
     ecnr: string;
     address: string;
     city: string;
     pin: string;
     district: string;
     country: string;
     nationality: string;
     email: string;
     
     created: Date;
     appUserIdNotEnforced: string;
     appUserId: number;
     notificationDesired: boolean;
     candidateStatus?: string;     //EnumCnadidateStatus.NotReferred;
     photoUrl?: string;
     lastActive?: Date;
     userPhones: IUserPhone[];
     userQualifications: IUserQualification[];
     userProfessions: IUserProfession[];
     userAttachments: IUserAttachment[];
     userExperiences: IUserExp[];
     userPhotos: IPhoto[];
     //userFormFiles: File[];
}

export class Candidate implements ICandidate {
     //checked: boolean;
     id= 0;
     applicationNo= 0;
     gender= '';
     firstName= '';
     secondName= '';
     familyName= '';
     knownAs= '';
     referredBy?= 0;
     referredByName= '';
     dOB?: Date;
     placeOfBirth= '';
     aadharNo= '';
     ppNo= '';
     ecnr='F';
     address= '';
     city= '';
     pin= '';
     district= '';
     country='India';
     nationality= '';
     email= '';
     companyId= 0;
     created= new Date;
     appUserIdNotEnforced= '';
     appUserId= 0;
     notificationDesired= false;
     candidateStatus?= '';     //EnumCnadidateStatus.NotReferred;
     photoUrl?= '';
     lastActive?= new Date;
     userPhones: IUserPhone[]=[];
     userQualifications: IUserQualification[]=[];
     userProfessions: IUserProfession[]=[];
     userAttachments: IUserAttachment[]=[];
     userExperiences: IUserExp[]=[];
     userPhotos: IPhoto[]=[];
     //userFormFiles: File[];
}